Adjustments for using TM and classic Tracking Code at the same time

MichaelHeerklotz opened the pull request:
* Added new setting "Use external Tracker" that allows to use a Matomo JS Tracker that is already loaded by the website by other means (for example the classic Matomo JS tracking code)
* Replaced Piwk.addTracker() with Piwik.getTracker(); this fixes using Matomo Tag Manager together with classic Matomo Code. addTracker() does some caching and did not work in my tests. getTracker() always creates a new Piwik instance, but since the Tag Manage code already caches the Piwik Instance this should be no problem

tsteur commented:
`Piwik.addTracker` had some bug that we'll fix shortly (it required idSite) to be set when it shouldn't. We'll fix that in core.
We don't really want to use `Piwik.getTracker` here as then you wouldn't be able to track events  or other things on the site using `_paq.push(['...'])` anymore. `getTracker` does not register an "async tracker" basically

MichaelHeerklotz commented:
Actually this is what I am trying to accomplish.
In my setup we have a lot of _paq.push() code that comes along with the classic Matomo Tracking Code. Now we want to add a Matomo Tag,  that should operate independant of the _paq.push() code (different site ids, different events, etc).

Now that you mention it, I see that it can make sense to support using _paq.push() when using Matomo Tag Manager, but I guess not many people are using this kind of setup (mixing tags and custom javascript tracking code). Maybe it makes sense to add an option to disable/enable this.

My users already tried to add the Matomo Tag Manager Js Code into their pages that already have the Classic Js Tracker Code, and assumed that it would work independently.

tsteur commented:
Setting to separate this behaviour sounds good I reckon.

So we have a setting to use `addTracker(default)` or `getTracker`.
* `addTracker` should be used when any `_paq.push()` in the website should also be applied to this tag.
* `getTracker` should be used when `_paq.push` should not affect the tracker in Matomo Tag.

Be good to explain this "nicely" in an understandable way :)

tsteur commented on Template/Tag/MatomoTag.web.js:
```diff
@@ -210,6 +210,10 @@
             // to initialize the tracker during window.piwikPluginAsyncInit
 
             var matomoConfig = parameters.get('matomoConfig', {});
+            if (matomoConfig.useExternalTracker) {
```
This one I'm not quite sure I understand?

MichaelHeerklotz replied:
Thank you for your feedback. I have merged this setting with the "bundleTracker" setting into a new "trackerSource" setting. 
trackerSource=external is for websites that already load matomo.js (asynchronously) and do not require the js tracker to be loaded by the container, while the container cannot detect it is already being loaded because it is loaded async.

MichaelHeerklotz replied:
Because the "bundleTracker" setting gets removed with this change, already existing configs will default back to  trackerSource=bundle, if you do not like this, we can change it

MichaelHeerklotz commented:
Thank you for your feedback. I have added a new setting "registerAsDefaultTracker" to switch between addTracker() and getTracker().

MichaelHeerklotz commented:
I am going to update this pr soon, so no need to merge it right now.

tsteur commented:
Awesome, let me know when you're done

mattab commented:
> Piwik.addTracker had some bug that we'll fix shortly (it required idSite) to be set when it shouldn't. 
> We'll fix that in core.
fyi this was fixed in https://github.com/matomo-org/matomo/pull/14787

MichaelHeerklotz commented:
@tsteur I have now updated the commit to include only the "registerAsDefaultTracker" settings.

I have dropped the other setting ("Use external Tracker") because it did not prove to be a stable way to load the tracker js.
